docs/src/main/sphinx/optimizer/cost-based-optimizations.rst

```diff
@@ -73,14 +72,38 @@ The valid values are:
 Capping replicated table size
 -----------------------------
 
-Join distribution type will be chosen automatically when join reordering strategy
-is set to ``AUTOMATIC`` or when join distribution type is set to ``AUTOMATIC``.
-In such case it is possible to cap the maximum size of replicated table via
-``join-max-broadcast-table-size`` config property (e.g. ``join-max-broadcast-table-size=100MB``)
-or via ``join_max_broadcast_table_size`` session property (e.g. ``set session join_max_broadcast_table_size='100MB';``)
-This allows to improve cluster concurrency and to prevent bad plans when CBO misestimates size of joined tables.
-
-By default replicated table size is capped to 100MB.
+Join distribution type will be chosen automatically when join reordering
+strategy is set to ``AUTOMATIC`` or when join distribution type is set to
+``AUTOMATIC``. In such cases, it is possible to cap the maximum size of the
+replicated table via the ``join-max-broadcast-table-size`` configuration
+property (e.g. ``join-max-broadcast-table-size=100MB``) or via the
+``join_max_broadcast_table_size`` session property (e.g.
+``set session join_max_broadcast_table_size='100MB';``). This allows you to
+improve cluster concurrency and to prevent bad plans when CBO misestimates size
+of joined tables.
+
+By default, replicated table size is capped to 100MB.
+
+Syntactic join order
+--------------------
+
+If not using cost-based optimization, Trino will default to syntactic join
+ordering. While there is no formal way to optimize queries for this case, it is
+possible to take advantage of how Trino implements joins to make them more
+performant.
+
+Trino uses in-memory `hash joins <https://en.wikipedia.org/wiki/Hash_join>`_.
+When performing joins, it will load the right-most table into memory, then
+stream the next right-most table to execute the join. If a query has multiple
+joins, the result of this first join will stay in memory, and the third
+right-most table will then be streamed, and so on for additional joins.
```

Comment:
The rest of this doc uses "build/probe" terminology to describe each side of a join, not the colloquial "right/left" terminology. If I'm reading this correctly, all of these should be "probe side" instead of "right-most table"